pyquil/analytics.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/python
+##############################################################################
+# Copyright 2016-2017 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+"""
+A collection of methods for analyzing pyquil programs (gate count, etc...)
+"""
+
+import pyquil.gates as gates
+from pyquil.quilbase import UnaryClassicalInstruction, BinaryClassicalInstruction, SimpleInstruction, QuilAction
+from grove.qft.fourier import qft
+from grove.unitary_circuit.arbitrary_state import create_arbitrary_state
+import numpy as np
+from pyquil.quil import Program
+from grove.teleport.teleportation import make_bell_pair
+import plotly
```

Comment:
We'd like to keep pyquil as lightweight as possible. We already introduced matplotlib as a plotting dependency for the QPU data.  We certainly should not have both plotly and matplotlib.  If plotly is clearly better, then we should replace the matplotlib code in the QPU plots.py file.

In particular, I think you should split this PR up into several.  At the least, one for the analysis code itself and a different one for plotting.  This is especially the case as the plotting one will include a change over from matplotlib to plotly (and some debate about which we should use)